packages/nuxt3/src/app/nuxt.ts

```diff
@@ -121,10 +121,10 @@ export function createNuxtApp (options: CreateOptions) {
 
 export async function applyPlugin (nuxtApp: NuxtApp, plugin: Plugin) {
   if (typeof plugin !== 'function') { return }
-  const result = await callWithNuxt(nuxtApp, () => plugin(nuxtApp))
-  if (result && typeof result === 'object') {
-    for (const key in result) {
-      nuxtApp.provide(key, result[key])
+  const { provide } = await callWithNuxt(nuxtApp, () => plugin(nuxtApp)) || {}
+  if (provide && typeof provide === 'object') {
```

Comment:
This condition also passes by `provide: null` if type safety should take into account but I'm wondering do we need the overhead of check if partially handling edge cases? 

```suggestion
  if (provide) {
```